pkg/oc/admin/policy/modify_roles.go

```diff
@@ -470,22 +509,54 @@ func (o *RoleModificationOptions) AddRole() error {
 	roleBinding.RoleRef.Namespace = o.RoleNamespace
 	roleBinding.RoleRef.Name = o.RoleName
 
+	// warn if binding to non-existent role
+	if o.RoleAccessor != nil && o.PrintErrf != nil {
+		if _, err := o.RoleAccessor.GetRole(roleBinding.RoleRef.Name); err != nil {
+			if kapierrors.IsNotFound(err) {
+				o.PrintErrf("Warning: role '%s' not found\n", roleBinding.RoleRef.Name)
+			} else {
+				o.PrintErrf("Warning: role '%s' may not exist; unknown error: %v\n", roleBinding.RoleRef.Name, err)
+			}
+		}
+	}
 	newSubjects := authorizationapi.BuildSubjects(o.Users, o.Groups)
 	newSubjects = append(newSubjects, o.Subjects...)
-
 subjectCheck:
 	for _, newSubject := range newSubjects {
+		if o.PrintErrf != nil {
+			var err error
+			switch newSubject.Kind {
+			case authorizationapi.ServiceAccountKind:
+				if o.ServiceAccountClient != nil {
+					_, err = o.ServiceAccountClient.ServiceAccounts(newSubject.Namespace).Get(newSubject.Name, metav1.GetOptions{})
+				}
+			case authorizationapi.UserKind, authorizationapi.SystemUserKind:
+				if o.UserClient != nil {
+					_, err = o.UserClient.User().Users().Get(newSubject.Name, metav1.GetOptions{})
+				}
+			case authorizationapi.GroupKind, authorizationapi.SystemGroupKind:
```

Comment:
system users and groups will never exist as API objects (name validation precludes that), so there's no point in looking those up

Reply:
I'll just remove them: bbdf4f9616